e2e/cypress/integration/channel/message_reply_pretext_spec.js

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import users from '../../fixtures/users.json';
+
+const sysadmin = users.sysadmin;
+
+describe('Message Reply with attachment pretext', () => {
+    let newChannel;
+    let botsUserId;
+    let accessToken;
+
+    before(() => {
+
+        // # Set ServiceSettings to expected values
+        const newSettings = {
+            ServiceSettings: {
+                EnableBotAccountCreation: true,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        // # Login and go to /
+        cy.apiLogin('sysadmin');
+
+        // # Create and visit new channel
+        cy.createAndVisitNewChannel().then((channel) => {
+            newChannel = channel;
+        });
+
+    });
+
+    it('MM-16734 Reply to an older bot post that has some attachment pretext', () => {
+        // # Get yesterdays date in UTC
+        const yesterdaysDate = Cypress.moment().subtract(1, 'days').valueOf();
+
+        const botName = 'bot-' + Date.now();
+
+        // # Create a bot and get userID
+        cy.apiCreateBot( botName,'Test Bot','test bot for E2E test replying to older bot post').then((userId) => {
+            botsUserId = userId;
+            const promoteToSysAdmin = (user) => {
+                cy.externalRequest({user: users.sysadmin, method: 'put', path: `users/botsUserId/roles`, data: {roles: 'system_user system_post_all'}});
+             };
+
+            //cy.apiAddUserToChannel(newChannel.id,botsUserId)
+            // # Get token from bots id
+            cy.apiAccessToken(botsUserId, "Create token").then((token) => {
+                accessToken = token;
+
+                //# Add bot to team
+                cy.apiAddUserToTeam(newChannel.team_id,botsUserId);
+                //# Add bot to channel
+                cy.apiAddUserToChannel(newChannel.id,botsUserId);
+                // # Post message with auth token
+                cy.apiPostBotMessage(newChannel.id, 'Hello message from '+ botName, 'Some Pretext', 'Some text', accessToken);
```

Comment:
Until this line, the user session is still as `sysadmin`.  That's why it's posted as sysadmin when `cy.apiPostBotMessage` is called.  I suggests to logout (`cy.apiLogout()`) before this line and log in again when verifying posted message.

Reply:
Okay, I've looked into this again and I think the better approach here is to use `cy.postMessageAs`, just need to modify it to accept token and message props.  With that, (externally) posting a message don't need to worry about the current session and so logging out is not necessary too.

Therefore, this can be:
```javascript
const message = 'Hello message from ' + botName;
const props = {attachments: [{pretext: 'Some Pretext', text: 'Some Text'}]};
cy.postMessageAs({token, message, props, channelId: newChannel.id});
```

[post_message_as.js](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/plugins/post_message_as.js) would be:
```diff
-module.exports = async ({sender, message, channelId, rootId, createAt = 0, baseUrl}) => {
-    const loginResponse = await axios({
-        url: `${baseUrl}/api/v4/users/login`,
-        headers: {'X-Requested-With': 'XMLHttpRequest'},
-        method: 'post',
-        data: {login_id: sender.username, password: sender.password},
-    });
+module.exports = async ({sender, token, message, props = {}, channelId, rootId, createAt = 0, baseUrl}) => {
+    const auth = {};
 
-    const setCookie = loginResponse.headers['set-cookie'];
-    let cookieString = '';
-    setCookie.forEach((cookie) => {
-        const nameAndValue = cookie.split(';')[0];
-        cookieString += nameAndValue + ';';
-    });
+    if (token) {
+        auth.Authorization = `Bearer ${token}`;
+    } else if (sender) {
+        const loginResponse = await axios({
+            url: `${baseUrl}/api/v4/users/login`,
+            headers: {'X-Requested-With': 'XMLHttpRequest'},
+            method: 'post',
+            data: {login_id: sender.username, password: sender.password},
+        });
+
+        const setCookie = loginResponse.headers['set-cookie'];
+        let cookieString = '';
+        setCookie.forEach((cookie) => {
+            const nameAndValue = cookie.split(';')[0];
+            cookieString += nameAndValue + ';';
+        });
+
+        auth.Cookie = cookieString;
+    }
 
     let response;
     try {
@@ -25,12 +33,13 @@ module.exports = async ({sender, message, channelId, rootId, createAt = 0, baseU
             headers: {
                 'Content-Type': 'application/json',
                 'X-Requested-With': 'XMLHttpRequest',
-                Cookie: cookieString,
+                ...auth,
             },
             method: 'post',
             data: {
                 channel_id: channelId,
                 message,
+                props,
                 type: '',
                 create_at: createAt,
                 parent_id: rootId,
```

then [postMessageAs](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/support/task_commands.js#L11) command would be:
```diff
-Cypress.Commands.add('postMessageAs', ({sender, message, channelId, rootId, createAt}) => {
+Cypress.Commands.add('postMessageAs', ({sender, token, message, props, channelId, rootId, createAt}) => {
     const baseUrl = Cypress.config('baseUrl');
 
-    cy.task('postMessageAs', {sender, message, channelId, rootId, createAt, baseUrl}).then(({status, data}) => {
+    cy.task('postMessageAs', {sender, token, message, props, channelId, rootId, createAt, baseUrl}).then(({status, data}) => {
         expect(status).to.equal(201);
```